gtfs-realtime/proto/gtfs-realtime.proto

```diff
@@ -271,6 +271,28 @@ message TripUpdate {
       // (e.g., `VehiclePosition.stop_id`).
       // NOTE: This field is still experimental, and subject to change. It may be formally adopted in the future.
       optional string assigned_stop_id = 1;
+      
+      // The updated headsign of the vehicle at the stop.
+      optional string stop_headsign = 2;
+      
+      // The updated pickup of the vehicle at the stop.
+      optional DropOffPickupType pickup_type = 3;
+      
+      // The updated drop off of the vehicle at the stop.
+      optional DropOffPickupType drop_off_type = 4;
+      
+      // The updated timing point status of the vehicle at the stop. 
+      // If false, the vehicle will no longer wait at the stop; 
+      // if true, the vehicle will wait until the scheduled time at the stop.
+      optional bool timepoint = 5;
+      
+      // Identifies the boarding booking rule at this stop time. 
+      // Refers to a `booking_rule_id` defined in the GTFS `booking_rules.txt`.
+      optional string pickup_booking_rule_id = 6;
```

Comment:
I have no idea if it should be possible to dynamically add a flex trip or not. Please advise.

Reply:
Since it's almost impossible to remove fields from the proto file, I only ever add those which I definitely have a need for. I don't add fields "to be on the safe side".

So if you don't see your self using it, leave it out.

Reply:
Yes, it's needed.

For example, an agency may run a short term relief which only calls at certain points when someone requests a ride.

Reply:
So you're keeping these booking rules around in case you need them in realtime updates?

Reply:
Yeah, the question is not _if_ it could be used, but is it actually being used. [Speculative features are discouraged. ](https://github.com/google/transit/blob/master/gtfs-realtime/CHANGES.md#speculative-features-are-discouraged)

Reply:
Do you know if there are any agencies which have added trips and require calling the agency to board or alight? If not I will remove the field from the PR.

Reply:
No I don't know any such service right now

Reply:
Yes, please remove it.

Reply:
I have removed it and only leave the fields that I will definitely use.